examples/python-shell.py

```diff
@@ -0,0 +1,85 @@
+import asyncio
+import sys
+import shlex
+
+from yapapi.payload import vm
+from yapapi.services import Service
+
+from service_manager import ServiceManager
+
+
+class PythonShell(Service):
+    @classmethod
+    async def get_payload(cls):
+        #   This image is created from examples/python-shell/provider/Dockerfile and pushed as described here:
+        #   https://handbook.golem.network/requestor-tutorials/convert-a-docker-image-into-a-golem-image
+        image_hash = "4c68e8d799acfa105a6b7b3553fcee2afa7f474b7aa4ce6a29f01d9e"
+        return await vm.repo(image_hash=image_hash)
+
+    async def start(self):
+        print(f"Initializing shell on provider {self.provider_name}, wait for prompt ...")
+        self._ctx.run('/bin/sh', '-c', 'nohup python shell.py run &')
+        yield self._ctx.commit()
+        print("Python shell is ready.")
+
+    async def run(self):
+        while True:
+            #   Print python shell output
+            self._ctx.run('/usr/local/bin/python', 'shell.py', 'read')
+            future_results = yield self._ctx.commit()
+            result = future_results.result()[-1]
+            print(result.stdout, end='', flush=True)
+
+            #   Pass command to shell
+            signal = await self._listen()
+            cmd = signal.message
+            cmd = shlex.quote(cmd)
+            self._ctx.run('/bin/sh', '-c', f"echo {cmd} | python shell.py write")
```

Comment:
Where is the `shell.py` file - is it a standard thing I'm just not aware of?

Reply:
This is in `examples/python-shell/provider` directory, that is not part of the PR because it was already merged.
It is part of the image.